client/tests/integration/permissions.rs

```diff
@@ -201,3 +209,83 @@ fn permissions_checked_before_transaction_execution() {
 
     assert!(root_cause.contains("Account does not have the needed permission token"));
 }
+
+#[test]
+fn permissions_differ_not_only_by_names() {
```

Comment:
Good testing style: a scenario is more important to test than specific behaviour.